python/GangaDirac/Lib/Utilities/DiracUtilities.py

```diff
@@ -272,11 +281,23 @@ def execute(command,
                                   eval_includes=eval_includes,
                                   update_env=update_env)
 
+    # If the time 
+    if returnable == 'Command timed out!':
+        raise GangaDiracException("DIRAC command timed out")
+
     # TODO we would like some way of working out if the code has been executed correctly
     # Most commands will be OK now that we've added the check for the valid proxy before executing commands here
 
     if cwd is None:
         shutil.rmtree(cwd_, ignore_errors=True)
 
-    return returnable
+    if isinstance(returnable, dict) and not return_raw_dict:
+        # If the output is a dictionary allow for automatic error detection
+        if returnable['OK']:
+            return returnable['Value']
+        else:
+            raise GangaDiracException(returnable['Message'])
+    else:
+        # If the output is NOT a dictionary return it
```

Comment:
Yes, if the user is talking to DIRAC or `print('foo')` or `output('foo')` as in the tests.